EIPS/eip-1532.md

```diff
@@ -0,0 +1,264 @@
+
+* @param debtID - is a unique identifier of a debt inside this contract space
+* @param newOwner - the address to make the receiver of all new debt
+
+```solidity
+function changeDebtContractOwner (uint256 debtID, address newOwner) public;
+```
+
+#### events
+
+##### Debt Created
+```solidity
+event DebtCreated(uint indexed debtID, uint indexed debtor, uint indexed creditor, uint amount);
+```
+
+##### Debt Fulfilled
+```solidity
+event DebtFulfilled(uint indexed debtID, uint indexed debtor, uint indexed creditor, uint amount);
+```
+
+##### Debtor Updated
+```solidity
+event DebtorUpdated(uint indexed debtID, uint indexed oldDebtor, uint indexed newDebtor uint creditor;
+```
+
+#### State
+
+##### DebtStatus
+
+This user-defined type is used to represent the state of a specific debt.
+
+Approved - An agreement has been made that will result in the debt creation
+Materialized - The creditor has upheld his end of the agreement and now the debtor is liable to fulfill his end
+Defaulted - The debtor did not uphold his end of the agreement in the agreed upon time
+Fulfilled - The debtor fulfilled his terms of the agreement
+
+```solidity
+enum DebtStatus { Approved, Materialized, Defaulted, Fulfilled }
+```
+
+The above interface should be implemented by all tokens that represent debt of any sort. Additionally, for secured debt and interest-bearing debt, we propose the following supplemental interfaces.
+
+### Secured Debt
+The Secured Standard is used in addition to the debt standard to represent collateralized debt.
+
+#### Functions
+##### depositCollateral
+
+This function should enable the deposit of collateral goods for this debt. If @param token is set to a valid address, it should attempt to call the transferFrom method of the token address and send @param amount of the tokens from the msg.sender to this smart contract. If msg.value is positive, it should mark msg.value as collateral for this specific debt. It should only allow collateral to be paid in one form of currency for each debt. (Accepted: one ERC20 token or Ether. Not Accepted: Two different types of ERC20 tokens, ERC20 token and Ether).
+
+* @param debtID - is a unique identifier of a debt inside this contract space
+* @param token - the address of the token used as collateral
+* @param amount - the amount of the token to be collateralized
+
+```solidity
+function depositCollateral(uint debtID, address token, uint amount) public payable;
+```
+
+##### claimCollateral
+
+If this method is called by the debtor, it should check if the debt has been fulfilled. If the debt is fulfilled, allow the msg.sender to withdraw the collateral. If this method is not called by the debtor, it should check if the msg.sender is the owner of the debt and transfer the collateral only if debtor defaulted.
+
+```solidity
+function claimCollateral() public;
```

Comment:
Shouldn't this have `debtID` as a parameter?

Reply:
Yes, you are correct. Thanks for the catch!